### DISCUSSION TITLE
cloudflare images

### DISCUSSION THREAD

**Discussion by skopy9581** [thumbsup/thumbsup#311] [Ideas]
Hi, it would be very interesting feature to integrate cloudflare images platform so you could upload html to cloudflare pages and have thumbsup upload to cf and generate links...

**Comment by rprieto** [comment]
Hi. How would you want to use Cloudflare images? Would you expect Cloudflare to do all the resizing, and Thumbsup to just look after the HTML?

**Reply by skopy9581** [reply]
Hi, yes, you could define variants before and have thumbsup upload/delete images with api and generate html
Tx

**Comment by skopy9581** [comment]
Hi, any news if this option is even considered to be implemented,

i would do it myself but simply lack  coding knowledge =(

**Comment by rprieto** [comment]
Thanks for adding more detail. I understand why it would be interesting not having to process images locally.

From a tech point of view, the challenges would be to make the image generation code swappable, and to reimplement all options on Cloudflare (resizing, configurable bitrate, metadata stripping, watermarks, etc).

From a conceptual point of view, it's also a challenge because thumbsups is first and foremost a gallery generator. It would have to become very clear when it's working locally, and when it produces HTML for remote images.

I like the idea, but I think this is outside the scope of Thumbsup for the moment. I'm happy to revisit this in the future though.
